rust/lance/src/datafusion/logical_plan.rs

```diff
@@ -0,0 +1,210 @@
+        vector_builder.values().append_value(0);
+        vector_builder.values().append_value(1);
+        vector_builder.append(true);
+        vector_builder.values().append_value(0);
+        vector_builder.values().append_value(1);
+        vector_builder.append(true);
+        vector_builder.values().append_value(2);
+        vector_builder.values().append_value(3);
+        vector_builder.append(true);
+        let vector = vector_builder.finish();
+
+        let utf8_values = StringArray::from(vec!["foo", "bar", "baz"]);
+
+        let batch = RecordBatch::try_new(
+            arrow_schema.clone(),
+            vec![Arc::new(vector), Arc::new(sa), Arc::new(utf8_values)],
+        )
+        .unwrap();
+        batches.push(batch);
+
+        (arrow_schema, batches)
+    }
+
+    #[tokio::test]
+    async fn test_dataset_logicalplan_projection_pd() {
+        let (schema, batches) = create_batches();
+        let test_dir = tempdir().unwrap();
+        let test_uri = test_dir.path().to_str().unwrap();
+
+        let batch_reader =
+            RecordBatchIterator::new(batches.clone().into_iter().map(Ok), schema.clone());
+
+        Dataset::write(batch_reader, test_uri, Some(WriteParams::default()))
+            .await
+            .unwrap();
+
+        let dataset = Dataset::open(test_uri).await.unwrap();
+        let ctx = SessionContext::new();
+        ctx.register_table("my_table", Arc::new(dataset)).unwrap();
+        let df = ctx.sql("SELECT vector, utf8 FROM my_table").await.unwrap();
+        let physical_plan = df.clone().create_physical_plan().await.unwrap();
+
+        assert!(physical_plan
+            .as_any()
+            .downcast_ref::<LanceScanExec>()
+            .is_some());
+
+        let expected_fields = schema
+            .fields()
+            .iter()
+            .filter_map(|f| {
+                if f.name() == "vector" || f.name() == "utf8" {
+                    Some(f.as_ref().clone())
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
+
+        let expected_schema = ArrowSchema::new(expected_fields);
+        assert_eq!(physical_plan.schema().as_ref(), &expected_schema);
+    }
+
+    #[tokio::test]
+    async fn test_dataset_logicalplan_struct_fields() {
+        let (schema, batches) = create_batches();
+        let test_dir = tempdir().unwrap();
+        let test_uri = test_dir.path().to_str().unwrap();
+
+        let batch_reader =
+            RecordBatchIterator::new(batches.clone().into_iter().map(Ok), schema.clone());
+
+        Dataset::write(batch_reader, test_uri, Some(WriteParams::default()))
+            .await
+            .unwrap();
+
+        let dataset = Dataset::open(test_uri).await.unwrap();
+        let ctx = SessionContext::new();
+        ctx.register_table("my_table", Arc::new(dataset)).unwrap();
+        let df = ctx
+            .sql("SELECT point.lat as lat, point.long as long FROM my_table")
```

Comment:
Do you know what expression type a struct field reference becomes? Been trying to figure that out.

Reply:
I guess GetIndexedField https://docs.rs/datafusion/latest/datafusion/logical_expr/struct.GetIndexedField.html

Reply:
Yeah it'll be `GetIndexedField`

```sql
select point.long from my_table
```
will get evaluated to this expr

```
Expr::GetIndexedField(
    GetIndexedField {
        expr: Column(
            Column {
                relation: Some(
                    Bare {
                        table: "my_table",
                    },
                ),
                name: "point",
            },
        ),
        field: NamedStructField {
            name: Utf8("long"),
        },
    },
)
```